bpo-30098: Adds verbose error to asyncio.ensure_future

crenwick opened the pull request:
Despite the [shy mention in the docs](https://docs.python.org/3/library/asyncio-task.html#asyncio.run_coroutine_threadsafe), it was not clear to me that the future returned from `asyncio.run_coroutine_threadsafe` is not compatible with `asyncio.ensure_future` (and other asyncio functions), and it took me a fair amount of frustration and source-code-digging to figure out what was going on.

To avoid this confusion for other users, I think that a verbose TypeError warning when a concurrent.futures.Future object is passed into `asyncio.ensure_future` would be very helpful.


Few questions:
 - Is there a better way to check if the obj is a concurrent.futures.Future?
 - Is the language in the TypeError ok? I tried to make it compatible with the default TypeError.

Let me know what you think! Thanks!

crenwick committed:
Adds verbose error to asyncio.ensure_future

Adds a verbose type error when a concurrent.futures.Future object is
passed into the asyncio.ensure_future function.

the-knights-who-say-ni commented:
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](http://bugs.python.org/) (b.p.o) to verify you have signed the CLA. This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://cpython-devguide.readthedocs.io/pullrequest.html#licensing) to rectify this issue.


Thanks again to your contribution and we look forward to looking at it!

Mariatta commented:
Hi, code changes like this aren't considered trivial. Please create an issue in the bug tracker https://bugs.python.org

Thanks :)

crenwick commented:
[Filed!](http://bugs.python.org/issue30098) (I also signed the CLA.)

Mariatta commented:
Thanks :) I changed the title to include the bpo-# 

The CLA can take one business day before it gets updated to your account.

crenwick committed:
Removes additional TypeError, changes existing language

Removes the verbose error that is raised only when a
concurrent.futures.Future object is passed into `ensure_future`, and
instead changes the language of the already existing TypeError to read
`asyncio.Future` instead of just `Future`

1st1 commented:
@crenwick Did you update your bpo profile with your GH username?

crenwick commented:
@1st1  [Yup!](http://bugs.python.org/user25981)

Mariatta commented:
Thanks @crenwick and congrats for your first contribution to CPython :tada:

I will backport this later today.
